qiskit/dagcircuit/dagdependency.py

```diff
@@ -567,6 +568,12 @@ def merge_no_duplicates(*iterables):
             yield val
 
 
+@lru_cache(maxsize=None)
```

Comment:
There is another use of `eye` in this fine:
```
id_op = np.reshape(np.eye(dim), (2, 2) * qbit_num)
```

Maybe it makes sense use this function there too? Maybe with a maxsize?